src/pages/Compare/index.tsx

```diff
@@ -62,43 +70,66 @@ export class CompareBase extends React.Component<Props> {
   }
 
   loadData(prevProps?: Props) {
-    const { match } = this.props;
+    const { match, path } = this.props;
     const { addonId, baseVersionId, headVersionId } = match.params;
 
     if (
       !prevProps ||
+      (prevProps.path && path !== prevProps.path) ||
       addonId !== prevProps.match.params.addonId ||
```

Comment:
This doesn't seem like it ever got test coverage:

```diff
diff --git a/src/pages/Compare/index.tsx b/src/pages/Compare/index.tsx
index 0e33d7a..e9926a1 100644
--- a/src/pages/Compare/index.tsx
+++ b/src/pages/Compare/index.tsx
@@ -76,7 +76,6 @@ export class CompareBase extends React.Component<Props> {
     if (
       !prevProps ||
       (prevProps.path && path !== prevProps.path) ||
-      addonId !== prevProps.match.params.addonId ||
       baseVersionId !== prevProps.match.params.baseVersionId ||
       headVersionId !== prevProps.match.params.headVersionId
     ) {
```

All tests pass

Reply:
ah oops, there is a test case for it but one parameter was missing :/